lib/AsyncStorage.js

```diff
@@ -52,10 +52,24 @@ type MultiRequest = {|
   reject: ?(error?: any) => void,
 |};
 
-function checkValueTypeNotString(value: any, usedKey: string) {
-  if (typeof value !== 'string') {
+function checkValidInput(usedKey: string, value: any) {
+  const isValuePassed = arguments.length > 1;
+
+  if (typeof usedKey !== 'string') {
+    console.warn(
+      `[AsyncStorage] Using ${typeof usedKey} type for key is not supported. This can lead to unexpected behavior/errors. Use string instead.\nKey passed: ${usedKey}\n`,
+    );
+  }
+
+  if (isValuePassed && value == null) {
```

Comment:
Should we nest this check inside the if-statement below?

Reply:
Yea, good catch